search-app/src/cmr/search/services/generic_association_service.clj

```diff
@@ -130,7 +130,12 @@
   associated_item: {concept_id: C6-PROV1}}."
   [mdb-context association operation]
   ;; save each association if there is no errors on it, otherwise returns the errors.
-  (let [association (sort-association association)
+  (let [associated-item-concept-id (:concept-id association)
+        revision-id (:revision-id association)
+        associated-item-revision-id (if (string? revision-id)
+                                      (read-string revision-id)
+                                      revision-id)
+        association (sort-association association)
         {source-concept-id :source-concept-id
```

Comment:
my question is what is important about the sort order here, this logic is done before the sorting, should a comment point out any details about why the order is significant?

Reply:
The side-effect with "sorting" is, if in the request I want to associate a service entry with a collection and a variable, in the two associations, the source is not always service entry. and the associated-item is not always collection and variable, the source and associated-item are switched in the service entry and collection association. The client still wants to know if the association with the collection is successful, but if your response says associated-item being service entry, then it doesn't give client any information.  This is why I need to get the original associated-item before the sorting.

Reply:
sort-function does explain why sorting is needed.